src/tests/test_viewsGenerationSystem.py

```diff
@@ -0,0 +1,174 @@
+		addon.addonVersion = MajorMinorPatch(0, 1)
+		addon.minNvdaAPIVersion = min(availableApiVersions)
+		addon.lastTestedVersion = max(availableApiVersions)
+		addon.channel = "stable"
+		_TestDataGenerator.write_mock_addon_to_files(addon, availableApiVersions)
+
+	@staticmethod
+	def generate_addon_with_versions():
+		"""Creates addon data to test the paths are correct across NVDA API Versions"""
+		# oldest version, to remain unlisted
+		addon = MockAddon()
+		addon.addonId = "multipleVersionsAddon"
+		addon.channel = "stable"
+		addon.addonVersion = MajorMinorPatch(2, 1)
+		addon.minNvdaAPIVersion = MajorMinorPatch(2020, 1)
+		addon.lastTestedVersion = MajorMinorPatch(2020, 4)
+		_TestDataGenerator.write_mock_addon_to_files(addon, {MajorMinorPatch(2020, 1), MajorMinorPatch(2020, 2)})
+
+		# newer version
+		addon = deepcopy(addon)
+		addon.addonVersion = MajorMinorPatch(13, 0)
+		addon.minNvdaAPIVersion = MajorMinorPatch(2020, 3)
+		addon.lastTestedVersion = MajorMinorPatch(2021, 1)
+		_TestDataGenerator.write_mock_addon_to_files(addon, {MajorMinorPatch(2021, 1)})
+
+
+class TestTransformation(unittest.TestCase):
+	@classmethod
+	def setUpClass(cls):
+		"""Empty the test data before the start of tests"""
+		if Path(DATA_DIR._root.value).exists():
+			shutil.rmtree(DATA_DIR._root.value)
+
+	def tearDown(self):
+		"""Empty the test data after each test"""
+		if Path(DATA_DIR._root.value).exists():
+			shutil.rmtree(DATA_DIR._root.value)
+
+	def _check_expected_addons_added(self):
+		"""
+		Checks that all the data and files in expectedResultsPath match the test output directory.
+		"""
+		testOutputFiles = set(glob.glob(DATA_DIR.OUTPUT + VIEWS_GLOB))
+		expectedFiles = set(glob.glob(DATA_DIR.EXPECTED_RES + VIEWS_GLOB))
+		testOutputFiles = set(f.lstrip(DATA_DIR.OUTPUT) for f in testOutputFiles)
+		expectedFiles = set(f.lstrip(DATA_DIR.EXPECTED_RES) for f in expectedFiles)
+		self.assertEqual(testOutputFiles, expectedFiles)
+		for outputFilename in testOutputFiles:
+			with open(f"{DATA_DIR.OUTPUT}/{outputFilename}", "r") as outputFile:
+				outputFileJson = json.load(outputFile)
+			with open(f"{DATA_DIR.EXPECTED_RES}/{outputFilename}", "r") as expectedFile:
+				expectedResultsJson = json.load(expectedFile)
+			self.assertDictEqual(expectedResultsJson, outputFileJson, msg=outputFilename)
+
+	def _test_transform(self) -> None:
+		"""Generates NVDA API versions so that transform can run.
+		"""
+		_TestDataGenerator.write_nvdaAPIVersions()
+		subprocess.run(
+			f"python -m src.transform {DATA_DIR.nvdaAPIVersionsPath} {DATA_DIR.INPUT} {DATA_DIR.OUTPUT}",
+			shell=True
+		).check_returncode()  # Raise CalledProcessError if the exit code is non-zero.
+
+	def test_transform_empty(self):
+		"""Confirms an empty transformation is successful
+		"""
+		self._test_transform()
+
+	def test_transform_successfully(self):
+		"""Confirms an transformation of a single addon is successful
+		"""
+		_TestDataGenerator.generate_addon()
+		self._test_transform()
```

Comment:
It would be good to be able to see what the addon data (in and out) looked like here. What about something like:
```suggestion
		_TestDataGenerator.generate_addons({
			"myAddonId/6.1.0":  addonMetaData("stable", requires="2020.1.0", tested="2021.2.0"),
			"myAddonId/6.2.0", addonMetaData("beta", requires="2020.3.0", tested="2021.2.0"),
			"myAddonId/6.2.0", addonMetaData("stable", requires="2020.3.0", tested="2021.2.0"),
			# and more to cover a few more cases
		)
		self.runTransform()
		self._check_output(expect={
			"2020.1/myAddonId/stable.json": "6.1.0",
			"2020.2/myAddonId/stable.json": "6.1.0",
			"2020.3/myAddonId/beta.json": "6.2.0",
			"2020.3/myAddonId/stable.json": "6.2.0,
			"2021.1/myAddonId/beta.json": "6.2.0",
			"2021.1/myAddonId/stable.json": "6.2.0,
			"2021.2/myAddonId/beta.json": "6.2.0",
			"2021.2/myAddonId/stable.json": "6.2.0,
		})
```